### ISSUE TITLE
Validation dataset are need?

### ISSUE THREAD

**Issue by sksksk1748** [wangguanan/Pytorch-Person-ReID-Baseline-PCB-Beyond-Part-Models#6]
Hi, Guan'an Wang

My English is very bad. I only use Google Translate to communicate. If I can communicate in Chinese, it’s best.

I have a ReID task problem for several days,
I think your code only uses train and test data sets,
No validation dataset is used.

I would like to ask your thoughts.
For ReID tasks, do you think validation is necessary?

According to the ReID tasks I have watched in the past few months,
some people are useful and some are not useful.
Hard to have a definite idea

I think for machine learning problems, validation is always important, since you need to make sure that your model is gaining generalization power instead of overfitting the test set. But since some areas still have very limited open data (still including ReID I guess?), the datasets used by academia don't have enough data for validation. With just a few validation data, the validation error might be pretty random and would not be very meaningful. So for research propose, I think it's fine to use just the training set and run massive experiments on serveral datasets for method comparison. But for enginners who aim to build a reliable model in the wild, it's necessary to collect a large amount of data and train the model with validation.

Have you ever thought about why you didn’t use the validation dataset when you wrote the program?

**Comment by wangguanan** [comment]
严格来说，机器学习算法都需要训练集训练模型，验证集调参数，测试集直接测试。

学术届ReID数据集比较小，所以一般就不再专门划分训练和验证集合。

**Comment by sksksk1748** [comment]
谢谢作者迅速的回覆我的问题
我是这个领域的小白，老师做的研究也不是这相关
CSDN、 知乎海外办帐号很难，也无法在相关文章发问
所以谢谢作者您让我可以透过 github 发问。

让我知道 ReID 这圈子的数据集不再专门划分训练和验证集合
那想请问在 ReID 这个圈子直接在 test dataset上调超参且換 model ，是普遍的做法吗? 
会这样问是因为我正在这样子做实验
我不禁思考这样子做的意义在哪
只是等一个"随机"最好的准确率很可能是 overfitting 的结果

祝周末愉快

**Comment by wangguanan** [comment]
直接在 test dataset 调参数确实有一些不合理，但是仍然能反应方法优劣。
优秀的方法经过简单调参可以达到比较好的效果，不好的方法即使各种调参也得不到好的结果。

当然，还是建议，首先把训练集划分为训练集和验证集合，然后使用验证集调参数。
最后，再用这套参数在完整的训练集训练。

**Comment by sksksk1748** [comment]
好，谢谢作者您的建议
对我的实验方法提供了一番说法
